wire-runtime/src/main/java/com/squareup/wire/internal/Internal.java

```diff
@@ -94,6 +133,17 @@ public static void checkElementsNotNull(List<?> list) {
     }
   }
 
+  /** Throw {@link NullPointerException} if {@code map} or one of its keys or values are null. */
+  public static void checkElementsNotNull(Map<?, ?> map) {
+    if (map == null) throw new NullPointerException("map == null");
+    if (map.containsKey(null)) throw new NullPointerException("map.containsKey(null)");
```

Comment:
probably simpler to just check in the loop?  You’re already going through entry-by-entry?